posts/2021-07-08-sig-bugs.adoc

```diff
@@ -0,0 +1,133 @@
+
+// end::intro[]
+
+
+// tag::run[]
+[#run]
+== Run your apps using 21.0.0.7
+
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty</groupId>
+    <artifactId>openliberty-runtime</artifactId>
+    <version>21.0.0.7</version>
+    <type>zip</type>
+</dependency>
+----
+
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[21.0.0.7,)'
+}
+----
+
+Or if you're using Docker:
+
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+
+//end::features[]
+
+[#bugs]
+== Notable bugs fixed in this release
+
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21.0.0.7+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.7].
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17344[Occasional login failure of OIDC RP if clientSecret is not configured]
++
+In the `OIDC RP`, when no `clientSecret` is configured, validation of the signature of the `JWT` occasionally would fail with a message similar to:
++
+[source, text]
+----
+CWWKS1739E: A signing key required by signature algorithm [none] was not available. null
+----
++
+This bug was fixed by updating the `JWT` validation code in the `OpenID` Connect client to return a `null` `java.security.Key` object as the signature verification key if `signatureAlgorithm` is set to `none` in the `OpenID` Connect client configuration. Before, an `HmacKey` object was instantiated using the `clientSecret` value from the `OIDC` client configuration, even though the Key object will never be used if `signatureAlgorithm` is set to `none`. The new behavior avoids a `NullPointerException` when a `clientSecret` is not configured in the `OIDC` client.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17593[EJB Singleton Lifecycle Deadlock]
++
+A Singleton can be locked from the lifecycle event causing a deadlock when:
+
+** When the Singleton called another managed object (like another session bean or asynchronous bean) OR the Singleton had a lifecycle interceptor.
+** The other managed object calls the Singleton. (Either through an asynchronous method that is invoked at the right time (so a timing problem) or through the method the Singleton called on it).
+
+For `@PreDestroy` we are no longer locking the singleton, depending on when the destroy resolves, the other managed object might get an error stating the bean has been stopped/destroyed.
```

Comment:
These 2 paragraphs aren't indented to align with the issue so they look out of place, I think the bulleted points above messed it up but idk how to fix it

Reply:
Done